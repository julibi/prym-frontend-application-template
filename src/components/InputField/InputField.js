import React from 'react';
import classNames from 'classnames';

import style from "./InputField.module.scss";

const InputField = ({ className, errorMessage, label, name, onChange, type = 'text', value }) => {
  return(
    <>
      <label
        className={style.label}
        htmlFor={name}  
      >
        {label}
      </label>
      <input
        className={classNames(style.inputField, className)}
        value={value}
        name={name}
        type={type}
        onChange={onChange}
      />
      <p className={style.errorMessage}>{errorMessage}</p>
    </>
  );
};

export default InputField;