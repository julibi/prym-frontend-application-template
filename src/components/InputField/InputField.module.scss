
.inputField {
  margin-top: 12px;
  height: 40px;
  width: 100%;
  font-size: large;
}

.errorMessage {
  color: #FF0000;
  font-weight: 400;
}