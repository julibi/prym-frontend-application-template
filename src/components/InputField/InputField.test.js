import InputField from './InputField';
import React from "react";
import { render, fireEvent } from "@testing-library/react";


describe('InputField', () => {
  const props = {
    label: 'label',
    name: 'name',
    onChange: jest.fn(),
    label: 'label',
    placeholder: 'test'
  };
  it('should render correctly with props', () => {
    const {container} = render(
      <InputField { ...props }/>
    );
    expect(container).toMatchSnapshot();
  });

  it('should render correctly with all optional props provided', () => {
    const newProps = { ...props, className: 'className', type: 'password', errorMessage: 'errorMessage' };
    const {container} = render(
      <InputField { ...newProps }/>
    );
    expect(container).toMatchSnapshot();
  });

  it('should fire onChange prop', () => {
    const { getByPlaceholderText, container } = render(<InputField {...props} />);
    const input = getByPlaceholderText('test');
    
    fireEvent.change(input, {
      target: {
        value: 'inputValue',
      },
    });

    expect(props.onChange).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
  });
});
