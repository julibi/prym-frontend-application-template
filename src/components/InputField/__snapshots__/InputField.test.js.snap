// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField should fire onChange prop 1`] = `
<div>
  <label
    class="label"
    for="name"
  >
    label
  </label>
  <input
    class="inputField"
    name="name"
    placeholder="test"
    type="text"
    value=""
  />
  <p
    class="errorMessage"
  />
</div>
`;

exports[`InputField should render correctly with all optional props provided 1`] = `
<div>
  <label
    class="label"
    for="name"
  >
    label
  </label>
  <input
    class="inputField className"
    name="name"
    placeholder="test"
    type="password"
    value=""
  />
  <p
    class="errorMessage"
  >
    errorMessage
  </p>
</div>
`;

exports[`InputField should render correctly with props 1`] = `
<div>
  <label
    class="label"
    for="name"
  >
    label
  </label>
  <input
    class="inputField"
    name="name"
    placeholder="test"
    type="text"
    value=""
  />
  <p
    class="errorMessage"
  />
</div>
`;
